packages/react/src/feature-app-container.tsx

```diff
@@ -148,13 +167,17 @@ class InternalFeatureAppContainer<
         {baseUrl, config, beforeCreate, done}
       );
 
-      if (!isFeatureApp(this.featureAppScope.featureApp)) {
+      const {featureApp} = this.featureAppScope;
+
+      if (!isFeatureApp(featureApp)) {
         throw new Error(
           'Invalid Feature App found. The Feature App must be an object with either 1) a `render` method that returns a React element, or 2) an `attachTo` method that accepts a container DOM element.'
         );
       }
 
-      this.state = {featureApp: this.featureAppScope.featureApp};
+      // TODO: schedule rerender with async ssr manager
+      // If no loading promise was given, "loading" should always always be false
+      this.state = {featureApp, loading: Boolean(featureApp.loadingPromise)};
```

Comment:
for ssr we should log a warning if loadingPromise is set and not set the state here. There is an `inBrowser` check in feature app loader, which can be reused here.